packaging/test_plugin_unified_storage_tiering.py

```diff
@@ -450,6 +452,47 @@ def test_put_and_get(self):
 
                     alice_session.assert_icommand('irm -f ' + filename)
 
+
+    def test_put_and_get_with_preserve_replica__92(self):
+        with storage_tiering_configured():
+            IrodsController().restart()
+            zone_name = IrodsConfig().client_environment['irods_zone_name']
+            with session.make_session_for_existing_admin() as admin_session:
+                with session.make_session_for_existing_user('alice', 'apass', lib.get_hostname(), zone_name) as alice_session:
+                    admin_session.assert_icommand('imeta add -R rnd0 irods::storage_tiering::preserve_replicas true')
+
+                    filename = "test_put_file"
+                    lib.create_local_testfile(filename)
+                    alice_session.assert_icommand('iput -R rnd0 ' + filename)
+                    alice_session.assert_icommand('imeta ls -d ' + filename, 'STDOUT_SINGLELINE', filename)
+                    alice_session.assert_icommand('ils -L ' + filename, 'STDOUT_SINGLELINE', filename)
+
+                    # wait for object to age out of tier 0
+                    sleep(5)
+
+                    # test stage to tier 1
+                    admin_session.assert_icommand('irule -r irods_rule_engine_plugin-unified_storage_tiering-instance -F /var/lib/irods/example_unified_tiering_invocation.r')
+                    wait_for_empty_queue(lambda: alice_session.assert_icommand('ils -L ' + filename, 'STDOUT_SINGLELINE', 'rnd0'))
+                    wait_for_empty_queue(lambda: alice_session.assert_icommand('ils -L ' + filename, 'STDOUT_SINGLELINE', 'rnd1'))
+
+                    # wait for object to age back out of tier 0
```

Comment:
what is the rest of this test doing?   i see it stages to 1, leaving a replica on 0... checks for both.

then ... i would expect it to stage 1 to 2 and then check for 0 and 2 and not-1?

what is the for loop doing in 487?

Reply:
Tier 2 is not interesting in this test.  The fact that we do not try to tier a preserved replica out of the Tier 0 is the point of this issue.